pinot-core/src/main/java/com/linkedin/pinot/core/data/GenericRow.java

```diff
@@ -53,18 +54,26 @@ public Object getValue(String fieldName) {
     return _fieldMap.get(fieldName);
   }
 
+  public void addField(String key, Object value) {
+    _fieldMap.put(key, value);
+  }
+
   @Override
   public String toString() {
     StringBuilder b = new StringBuilder();
     for (String key : _fieldMap.keySet()) {
-      if (_fieldMap.get(key) instanceof Object[]) {
-        b.append(key + " : " + Arrays.toString((Object[]) _fieldMap.get(key)) + ", ");
+      Object value = _fieldMap.get(key);
+      b.append(key);
+      b.append(" : ");
+      if (value instanceof Object[]) {
+        b.append(Arrays.toString((Object[]) value));
+      } else if (value instanceof HyperLogLog) {
+        b.append(((HyperLogLog) value).cardinality());
```

Comment:
still going through the review...will the key for HLL column indicate this is distinct count ?